draft-ietf-quic-qlog-main-schema.md

```diff
@@ -1280,182 +1294,124 @@ SimulationMarker = {
-".sqlog" (for "streaming" qlog) and the Media Type (if any) SHOULD be
-"application/qlog+json-seq" per {{!RFC8091}}.
+To overcome the issues related to JSON streaming, a qlog mapping to a streamable
+JSON format called JSON Text Sequences (JSON-SEQ) ({{!RFC7464}}) is provided.
 
 JSON Text Sequences are very similar to JSON, except that JSON objects are
 serialized as individual records, each prefixed by an ASCII Record Separator
 (\<RS\>, 0x1E), and each ending with an ASCII Line Feed character (\n, 0x0A). Note
 that each record can also contain any amount of newlines in its body, as long as
 it ends with a newline character before the next \<RS\> character.
 
-Each qlog event is serialized and interpreted as an individual JSON Text Sequence
-record, and can simply be appended as a new object at the back of an event stream
-or log file. Put differently, unlike default JSON, it does not require a file to
-be wrapped as a full object with "{ ... }" or "\[... \]".
-
-For this to work, some qlog definitions have to be adjusted however.
-Mainly, events are no longer part of the "events" array in the Trace
-object, but are instead logged separately from the qlog "header", as
-indicated by the TraceSeq object in {{trace-seq-def}}. Additionally,
-qlog's JSON-SEQ mapping does not allow logging multiple individual
-traces in a single qlog file. As such, the QlogFile:traces field is
-replaced by the singular QlogFileSeq:trace field, see
-{{qlog-file-seq-def}}. An example can be seen in {{json-seq-ex}}. Note
-that the "group_id" field can still be used on a per-event basis to
-include events from conceptually different sources in a single JSON-SEQ
-qlog file.
+In order to leverage the streaming capability, each qlog event is serialized and
+interpreted as an individual JSON Text Sequence record, that is appended as a
+new object to the back of an event stream or log file. Put differently, unlike
+default JSON, it does not require a document to be wrapped as a full object with
+"{ ... }" or "\[... \]".
+
+This alternative record streaming approach cannot be accommodated by QlogFile
+({{qlog-file-def}}). Instead, QlogFileSeq is defined in {{qlog-file-seq-def}},
+which notably includes only a single trace (TraceSeq) and omits an explicit
+events field. An example is provided in {{json-seq-ex}}.
+
+When mapping qlog to JSON-SEQ, the "qlog_format" field MUST have the value
+"JSON-SEQ". Rhe file extension/suffix SHOULD be ".sqlog" (for "streaming" qlog).
+The Media Type, if any, SHOULD be "application/qlog+json-seq" per {{!RFC8091}}.
+
+While not specifically required by the JSON-SEQ specification, all qlog
+field names in a JSON-SEQ serialization MUST be lowercase.
+
+In order to serialize all other CDDL-based qlog event and data structure
+definitions to JSON-SEQ, the official CDDL-to-JSON mapping defined in
+{{Appendix E of CDDL}} SHOULD be employed.
+
+Note that the "group_id" field can still be used on a per-event basis to include
```

Comment:
Imo this should be moved up to right after the Trace vs TraceSeq discussion (line 1400 above, "which notably includes only a single trace (TraceSeq) and omits an explicit events field")